### DISCUSSION TITLE
Simulation consumed time

### DISCUSSION THREAD

**Discussion by jinningwang** [CURENT/andes#423] [Q&A] [answered]
Hi Hantao,

In a loop structure to run TDS, I found that the consumed time is linearly increased as shown below:
![image](https://user-images.githubusercontent.com/79226045/224766002-f90c40cc-a88d-47bd-ad81-ceccf424243b.png)

Is it possible to keep a relatively constant speed?

Part of the source code:
```python
case3 = os.path.join(path_case, 'npcc_htb.xlsx')

# --- set case ---
ss = andes.load(case3,
                no_output=True,
                default_config=False,
                setup=False)

ss.TDS.config.no_tqdm = 1  # turn off tqdm progress bar
ss.TDS.config.criteria = 0  # turn off stability criteria
ss.TDS.save_mode = 'manual'  # turn off auto save
ss.TDS.save_every = 0

ss.PFlow.run()  # solve power flow

t_total = 300
t_step = 0.05

size = int((t_total - ss.TDS.config.tf) / t_step + 1)
t_sim = np.zeros(size)
ctn = 0

p_htb = 0.1

while ss.TDS.config.tf < t_total:
    t0, _ = elapsed()
    ss.PQ.set(src='Ppf', idx=pq_htb, attr='v', value=p0 + p_htb)
    ss.TurbineGov.set(src='paux0', idx=tg_idx, attr='v', value=p_htb)
    ss.TDS.config.tf += t_step
    ss.TDS.run()
    t0, s0 = elapsed(t0)
    t_sim[ctn] = float(s0.split(' ')[0])
    ctn += 1
```

**Comment by jinningwang** [comment, marked as answer]
After further investigation, I added a parameter setting ``ss.TDS.config.limit_store = 1`` and then the consumed time is relatively constant as shown below:
![image](https://user-images.githubusercontent.com/79226045/224768855-3f424f85-194f-4fde-bc40-607600db7f17.png)

Thanks,
Jinning

**Reply by cuihantao** [reply]
Yeah, the extra time is for memory operation. When appending results `TDS.ts`'s storage, memory copy happens. When data grows, memory copy time gets longer.
